app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,23 @@
+module SessionsHelper
+  def log_in(user)
+    session[:user_id] = user.id
+  end
+
+  def current_user
+    @current_user ||= User.find_by(id: session[:user_id])
+    #prevents having to hit the database every time current_user is called on a page
+  end
+
+  def logged_in?
+    !current_user.nil?
+  end
+
+  def owns_profile?(user)
```

Comment:
This would be better named somethign like "is_same_user" as that's what it does. Having said that, this kind of is_owned_by or is_editable_by check can often be put into a model / or an activemodel concern.